src/System.Private.Xml/src/System/Xml/Serialization/Types.cs

```diff
@@ -1194,6 +1194,14 @@ private static bool ShouldBeReplaced(MemberInfo memberInfoToBeReplaced, Type der
                             replacedInfo = info;
                             if (replacedInfo != memberInfoToBeReplaced)
                             {
+                                if(memberInfoToBeReplaced is PropertyInfo propertyInfoToBeReplaced)
+                                {
+                                    if (!info.GetMethod.IsPublic && propertyInfoToBeReplaced.GetMethod.IsPublic)
```

Comment:
We probably can check `!info.GetMethod.IsPublic` first before checking `memberInfoToBeReplaced is PropertyInfo propertyInfoToBeReplaced`. In most cases, `info.GetMethod.IsPublic` would be true and thus we can avoid the other checks.